pkg/lib/oauthrelyingparty/oauthrelyingpartyutil/oidc.go

```diff
@@ -50,7 +51,7 @@ func FetchOIDCDiscoveryDocument(ctx context.Context, client *http.Client, endpoi
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to fetch OIDC discovery document: unexpected status code: %d", resp.StatusCode)
+		return nil, slogutil.ErrOIDCDiscoveryInvalidStatusCode
```

Comment:
Actually what we are missing is a general mechanism to collect metrics on the outgoing HTTP request we send to the external service. Ideally, we should track https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#http-client

So I suggest we just leave this alone first.